### PR TITLE
Auto OCP-18994 and OCP-18998

### PR DESCRIPTION
**Pull request by xiuwang**
@openshift/devexp-qe please help review, thanks

### CODE REVIEW

File: lib/rules/cli/4.1.yaml [hunk 1]
```diff
@@ -397,6 +397,19 @@
     :l: -l <value>
     :resource-names-file: --resource-names-file=<value>
     :svc_name: <value>
+:image_mirror:
+  :cmd: oc image mirror
```
Comment (wewang58) [inline lib/rules/cli/4.1.yaml]: maybe it's better    :cmd: oc image mirror <src> <des>
other LGTM

Comment (xiuwang) [reply lib/rules/cli/4.1.yaml]: @wewang58 Updated, thanks

File: features/step_definitions/registry.rb [hunk 2]
```diff
@@ -397,3 +397,44 @@
   cb_name ||= :generation_number
   cb[cb_name] = deployment(name).generation_number(user: user, cached: false)
 end
+
+Given /^docker config for default image registry is saved$/ do
```
Comment (pruan-rht) [inline features/step_definitions/registry.rb]: @xiuwang the step name is not very clear where `docker config` is saved to, please be more specific when the step name.

File: features/step_definitions/registry.rb [hunk 3]
```diff
@@ -397,3 +397,44 @@
...
+  File.open("#{cb.dockercfg_file}", 'wb') { |f| 
+    f.write(
+       {   
+         "auths" => generated_cfg
+       }.to_json
+    )
+  }
+end
+
+Given /^certification for default image registry is saved$/ do
```
Comment (pruan-rht) [inline features/step_definitions/registry.rb]: @xiuwang same comment above

File: features/step_definitions/registry.rb (hunk 2 above)
Comment (xiuwang) [reply features/step_definitions/registry.rb]: @pruan-rht Thanks your advice, I have addressed this comment, please help review again.

File: features/step_definitions/registry.rb [hunk 4]
```diff
@@ -397,3 +397,45 @@
   cb_name ||= :generation_number
   cb[cb_name] = deployment(name).generation_number(user: user, cached: false)
 end
+
+Given /^docker config for default image registry is stored to the#{OPT_SYM} clipboard$/ do |cb_name|
+
+  step %Q/I enable image-registry default route/
+  step %Q/default image registry route is stored in the :integrated_reg_ip clipboard/
+  step %Q/a 5 characters random string of type :dns is stored into the :dockercfg_name clipboard/
+  cb[cb_name] ="/tmp/#{cb.dockercfg_name}"
```
Comment (pruan-rht) [inline features/step_definitions/registry.rb]: @xiuwang thanks for the update, but this step will fail if `cb_name` is not specified which will become `nil`.  You should add this a line like `cb_name ||= :dockercfg_name` Here's a good article if that explains what `||=` does http://www.rubyinside.com/what-rubys-double-pipe-or-equals-really-does-5488.html

File: features/step_definitions/registry.rb [hunk 5]
```diff
@@ -397,3 +397,45 @@
...
+       }.to_json
+    )
+  }
+end
+
+Given /^certification for default image registry is stored to the#{OPT_SYM} clipboard$/ do |cb_name|
+  ensure_admin_tagged
+  
+  step %Q/a 5 characters random string of type :dns is stored into the :regcrt clipboard/
+  cb[cb_name] ="/tmp/#{cb.regcrt}.crt"
```
Comment (pruan-rht) [inline features/step_definitions/registry.rb]: @xiaocwan same comment as above

File: features/step_definitions/registry.rb (hunk 4 above)
Comment (xiuwang) [reply features/step_definitions/registry.rb]: @pruan-rht Thanks, Peter~ Have updated all comments

**Comment by xiuwang** [comment]
@pruan-rht @akostadinov Could you help review this again or merge it?
Thanks

**Pull request merged**